src/main/java/uk/gov/digital/ho/hocs/audit/export/ExportService.java

```diff
@@ -333,6 +337,54 @@ void correspondentExport(LocalDate from, LocalDate to, OutputStreamWriter output
         }
     }
 
+    void extensionExport(LocalDate from, LocalDate to, OutputStreamWriter outputWriter, String caseTypeCode, boolean convert, boolean convertHeader, final ZonedDateTimeConverter zonedDateTimeConverter) throws IOException {
+        log.info("Exporting EXTENSION to CSV", value(EVENT, CSV_EXPORT_START));
+        List<String> headers = Stream.of("timestamp", "event", "userId", "caseId",
+                "created", "type", "note").collect(Collectors.toList());
+
+        List<String> substitutedHeaders = headers;
+        if (convertHeader) {
+            substitutedHeaders = headerConverter.substitute(headers);
+        }
+
+        try (CSVPrinter printer = new CSVPrinter(outputWriter, CSVFormat.DEFAULT.withHeader(substitutedHeaders.toArray(new String[substitutedHeaders.size()])))) {
+            Stream<AuditData> data = auditRepository.findAuditDataByDateRangeAndEvents(LocalDateTime.of(
+                    from, LocalTime.MIN), LocalDateTime.of(to, LocalTime.MAX),
+                    EXTENSION_EVENTS, caseTypeCode);
+
+            ExportDataConverter exportDataConverter = convert ? exportDataConverterFactory.getInstance() : null;
+
+            data.forEach((audit) -> {
+                try {
+                    String[] auditRow = getExtensionAuditPayloadRow(audit, zonedDateTimeConverter);
+                    if (convert){
+                        auditRow = exportDataConverter.convertData(auditRow, caseTypeCode);
+                    }
+                    auditRow = malformedDateConverter.correctDateFields(auditRow);
+                    printer.printRecord(auditRow);
+                    outputWriter.flush();
+                } catch (IOException e) {
+                    log.error("Unable to get record for audit {} for reason {}", audit.getUuid(), e.getMessage(), value(LogEvent.EVENT, CSV_EXPORT_FAILURE));
+                }
+            });
+            log.info("Export EXTENSION to CSV Complete", value(EVENT, CSV_EXPORT_COMPETE));
+        }
+    }
+
+    private String[] getExtensionAuditPayloadRow(AuditData audit, final ZonedDateTimeConverter zonedDateTimeConverter) throws IOException {
```

Comment:
Can you change the name of the method to match the pattern used throughout this module please? i.e. `parseExtensionAuditPayload`

Reply:
I have addressed this